addons/account_voucher/models/account_voucher.py

```diff
@@ -316,6 +331,16 @@ def action_move_line_create(self):
             # Create one move line per voucher line where amount is not 0.0
             line_total = voucher.with_context(ctx).voucher_move_line_create(line_total, move.id, company_currency, current_currency)
 
+            # Create a payment to allow the reconciliation when pay_now = 'pay_now'.
+            if self.pay_now == 'pay_now' and self.amount > 0:
+                payment_id = self.env['account.payment'].create(self.voucher_pay_now_payment_create())
+                payment_id.post()
+
+                # Reconcile the receipt with the payment
+                (payment_id.move_line_ids + move.line_ids)\
+                    .filtered(lambda l: l.account_id == self.account_id)\
+                    .reconcile()
```

Comment:
arf, it's correct but it hurts my eyes... use a variable and write that in 2 one-liners